tsdb/head.go

```diff
@@ -572,32 +636,216 @@ func (h *Head) Init(minValidTime int64) error {
 		return nil
 	}
 
+	start := time.Now()
 	level.Info(h.logger).Log("msg", "Replaying WAL, this may take awhile")
+
+	level.Info(h.logger).Log("msg", "creating series from WAL")
+	var err error
+	h.refSeries, h.multiRef, err = h.loadSeriesFromWAL()
+	if err != nil {
+		return err
+	}
+
+	// No corruption in loading series. Hence set these to nil to mark that.
```

Comment:
Would you mind if we use a dedicated variable with a more descriptive name for this.
or can try to use the metric to detect/indicate if there was a repair.

Reply:
I will use a dedicated variable to indicate this, we need this info in `PostWALRepairRecovery`

Reply:
so if you have a dedicated variable why do you still need to set `refSeries` and `multiRef` to nil?

also what do you mean by 

`Hence set these to nil to garbage collect it.`

Reply:
We need `refSeries` for perf reasons: not to load series by holding all the locks again.
We need to hold `multiRef` because that is created only when we replay series and we need it to replay samples properly. I have added an explanation in the comments here https://github.com/prometheus/prometheus/blob/d5e7aed61e4eedbd7daf48a5f204433051ea0efc/tsdb/head.go#L94-L96

> Hence set these to nil to garbage collect it.

Go will garbage collect the memory for the items which are not referenced anywhere in code currently (IIUC), so if we don't set it to nil, it will always be held in memory. Setting it to `nil` allows Go GC to collect it. I will mention Go GC in comment there, as we have other GC as well.

Reply:
Yes I was confused that you are talking about the internal garbage collection in the code

Why reason why this wouldn't work in the `PostWALRepairRecovery`>

```
if !h.seriesReplayCorruption {
	return nil
}
``

Reply:
It will, I will change it